### PR TITLE
[VAULT-20073] Docs: update docs on Sentinel RGP group policy application

### PR DESCRIPTION
**Pull request by kubawi**
Docs update companion to #23090 (Ensure Role Governing Policies are only applied down the namespace hierarchy).

### CODE REVIEW

File: website/content/docs/enterprise/sentinel/index.mdx [hunk 1]
```diff
@@ -87,15 +87,31 @@ a step-by-step instruction.
 
 </Tip>
 
-Consider the following scenario. 
+<Warning>
+  Starting with Vault versions 1.15.0, 1.14.4, and 1.13.8 RGPs derived
+  from identity group memberships are applied to requests in child namespaces,
+  instead of parent ones. Read the scenarios below to familiarize yourself with the changes.
```
Comment (VioletHynes) [inline website/content/docs/enterprise/sentinel/index.mdx:93]:
```suggestion
  Starting with Vault versions 1.15.0, 1.14.4, and 1.13.8 RGPs derived
  from identity group memberships are applied to entities in child namespaces,
  only. Read the scenarios below to familiarize yourself with the changes.
```

Comment (VioletHynes) [reply website/content/docs/enterprise/sentinel/index.mdx:93]: I believe this is correct: if I'm in the root namespace, and a child imposed a deny-all to me, I would still be able to read the child namespace, as I understand it.

Comment (kubawi) [reply website/content/docs/enterprise/sentinel/index.mdx]: Thanks, Violet!

**Review by VioletHynes — APPROVED** [review]
Looks good! Thanks Kuba!

**Review by peteski22 — APPROVED** [review]
Used the [PR Review comment key](https://hashicorp.atlassian.net/wiki/spaces/VAULT/pages/2545254986/PR+Review+comment+key)

File: website/content/api-docs/system/config-group-policy-application.mdx [hunk 2]
```diff
@@ -10,6 +10,12 @@ description: The '/sys/config/group-policy-application' endpoint is used to conf
 
 @include 'alerts/restricted-root.mdx'
 
+<Warning>
+  Starting with Vault versions `1.15.0`, `1.14.4`, and `1.13.8` the group policy application mode
+  setting applies only to ACL policies. Vault Sentinel Role Governing Policies (RGPs) are not affected
+  by group policy application mode.
```
Comment (schavis) [inline website/content/api-docs/system/config-group-policy-application.mdx:16]:
```suggestion
  The group policy application mode only applies to ACL policies and no longer
  affects Vault sentinel role governing policies (RGPs) for the following Vault
  versions:

  - `1.15.0+`
  - `1.14.4+`
  - `1.13.8+`

```

File: website/content/docs/enterprise/sentinel/index.mdx [hunk 3]
```diff
@@ -87,15 +87,31 @@ a step-by-step instruction.
 
 </Tip>
 
-Consider the following scenario. 
+<Warning>
+  Starting with Vault versions 1.15.0, 1.14.4, and 1.13.8 RGPs derived
+  from identity group memberships are applied only to entities in child namespaces. 
+  Read the scenarios below to familiarize yourself with the changes.
+</Warning>
```
Comment (schavis) [inline website/content/docs/enterprise/sentinel/index.mdx:94]:
```suggestion
<Warning>

  As of the following versions, Vault only applies RPGs derived from identity
  group membership to entities in child namespaces:

  - `1.15.0+`
  - `1.14.4+`
  - `1.13.8+`

</Warning>

The scenarios below describe the relevant changes in more detail.
```

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/api-docs/system/config-group-policy-application.mdx (hunk 2 above)
Comment (kubawi) [reply website/content/api-docs/system/config-group-policy-application.mdx:16]: Thanks, Sarah!

**Review by schavis — APPROVED** [review]

**Comment by schavis** [comment]
@kubawi Don't forget to add the appropriate backport labels to publish the doc updates to the current (1.15) and any previous docs that should reflect the changes.

**Comment by kubawi** [comment]
> @kubawi Don't forget to add the appropriate backport labels to publish the doc updates to the current (1.15) and any previous docs that should reflect the changes.

Will do. Thanks, Sarah!

**Pull request merged**